### PR TITLE
♻️On-demand clusters: remove dask-gateway

### PR DESCRIPTION
**Pull request by sanderegg**
#### Clusters-keeper
- [x] the clusters-keeper deployed machines now start a dask-scheduler/dask-worker directly instead of osparc-dask-gateway
  - deploys a docker-compose.yml stack in the machine,
  - the dask-sidecar is set as global
- [x] removed dask-gateway/dask-gateway-server dependencies in clusters-keeper
- [x] moved dask-scheduler fixture to pytest-simcore (shared between dv-2 and clusters-keeper, and soon autoscaling)
- [x] @mguidon the name of the machine created on EC2 was modified to be more user-friendly, it now goes along: ```osparc-computational-cluster-manager-{app_settings.SWARM_STACK_NAME}-user_id:{user_id}-wallet_id:{wallet_id}```
- [x] add a distributed lock mechanism to prevent creating multiple clusters for the same user_id/wallet_id pair if the RPC is called in a short burst
- [x] fixed pipeline cannot be stopped when *waiting for cluster*

#### dask-sidecar
- [x] scheduler now also shows a logo on start
- [x] fixed issue in devel mode where ```/home/scu/.config/dask/distributed.yml``` was not writable

### director-v2
- [x]   adapted code to change from dask-gateway to dask-scheduler

### CODE REVIEW

File: services/clusters-keeper/requirements/_base.in
Comment (pcrespov) [inline services/clusters-keeper/requirements/_base.in:1]: Q: what about `services/osparc-gateway-server`? you are going to delete it?

File: services/clusters-keeper/src/simcore_service_clusters_keeper/_meta.py [hunk 1]
```diff
@@ -35,7 +36,11 @@ def get_summary() -> str:
 
 
 SUMMARY: Final[str] = get_summary()
-
+PACKAGE_DATA_FOLDER: Final[Path] = Path(
```
Comment (pcrespov) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/_meta.py:39]: Not going to adapt packages/service-library/src/servicelib/utils_meta.py and use it here??

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py [hunk 2]
```diff
@@ -49,42 +19,34 @@ async def _wrap_client_async_routine(
...
-            _logger.info("no cluster in gateway, nothing going on")
-            return False
-        assert len(cluster_reports) == 1  # nosec
-        async with gateway.connect(
-            cluster_reports[0].name, shutdown_on_close=False
-        ) as dask_cluster, dask_cluster.get_client() as client:
-            datasets_on_scheduler = await _wrap_client_async_routine(
-                client.list_datasets()
-            )
+async def ping_scheduler(url: AnyUrl) -> bool:
```
Comment (pcrespov) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:22]: THOUGHT: at some point i would like to distinguish health-checks at the transport level and at the application level. I assume ping is the former? i.e. the container is reachable but the app still not fully initialized? Do we really need both?

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py [hunk 3]
```diff
@@ -49,42 +19,34 @@ async def _wrap_client_async_routine(
...
+    try:
+        async with distributed.Client(f"{url}", asynchronous=True, timeout="5"):
+            ...
```
Comment (pcrespov) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:25]: MINOR: this context manager with nothing inside read very strange ...

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py [hunk 4]
```diff
@@ -49,42 +19,34 @@ async def _wrap_client_async_routine(
...
+        _logger.info(
+            "osparc-dask-scheduler %s ping timed-out, the machine is likely still starting...",
+            url,
+        )
+
+    return False
+
+
+async def is_scheduler_busy(url: AnyUrl) -> bool:
+    async with distributed.Client(f"{url}", asynchronous=True) as client:
```
Comment (pcrespov) [inline services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:37]: MINOR: `AnyUrl` is already a `str` right?

File: services/dask-sidecar/src/simcore_service_dask_sidecar/_meta.py [hunk 5]
```diff
@@ -16,15 +16,31 @@
...
+ | (_) \__ \ |_) | (_| | | | (__  | (_| | (_| \__ \   <  \__ \ | (_| |  __/ (_| (_| | |
+  \___/|___/ .__/ \__,_|_|  \___|  \__,_|\__,_|___/_|\_\ |___/_|\__,_|\___|\___\__,_|_|             v{__version__} with dask=={dask.__version__}
+           |_|
+"""
+
+DASK_SCHEDULER_APP_STARTED_BANNER_MSG = rf"""
+
+                                       _           _               _              _       _
+   ___  ___ _ __   __ _ _ __ ___    __| | __ _ ___| | __  ___  ___| |__   ___  __| |_   _| | ___ _ __
+  / _ \/ __| '_ \ / _` | '__/ __|  / _` |/ _` / __| |/ / / __|/ __| '_ \ / _ \/ _` | | | | |/ _ \ '__|
```
Comment (pcrespov) [inline services/dask-sidecar/src/simcore_service_dask_sidecar/_meta.py:33]: OMG! :-D

**Review by pcrespov — APPROVED** [review]
Except for the "unfancy" banner ;-) ... just some minor comments

File: services/clusters-keeper/requirements/_base.in
Comment (sanderegg) [reply services/clusters-keeper/requirements/_base.in:1]: not yet. it is still in use for user-controlled gateways

File: services/clusters-keeper/src/simcore_service_clusters_keeper/_meta.py (hunk 1 above)
Comment (sanderegg) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/_meta.py:39]: My usual, is to move things once I need them in at least 2 services...

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py (hunk 3 above)
Comment (sanderegg) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:25]: hmm, how would you actually do it? I'm checking if I can connect the client to the scheduler.

File: services/dask-sidecar/src/simcore_service_dask_sidecar/_meta.py (hunk 5 above)
Comment (sanderegg) [reply services/dask-sidecar/src/simcore_service_dask_sidecar/_meta.py:33]: isn't it amazing

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py (hunk 4 above)
Comment (sanderegg) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:37]: you are actually right

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py (hunk 2 above)
Comment (sanderegg) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:22]: so here the ping_scheduler tries to connect with the dask-scheduler.
Even chatty just proposed the same name... let me know what you prefer

**Review by YuryHrytsuk — APPROVED** [review]

**Review by matusdrobuliak66 — APPROVED** [review]

File: services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py (hunk 3 above)
Comment (pcrespov) [reply services/clusters-keeper/src/simcore_service_clusters_keeper/modules/dask.py:23]: It is perfectly correct but simply "reads strange" to have a context manage with "nothing to run inside".

**Review by GitHK — APPROVED** [review]

**Pull request merged**